ringbuf/reader.go

```diff
@@ -204,22 +204,21 @@ func (r *Reader) ReadInto(rec *Record) error {
 	for {
 		if !r.haveData {
 			_, err := r.poller.Wait(r.epollEvents[:cap(r.epollEvents)], r.deadline)
-			if err != nil {
+			if errors.Is(err, os.ErrClosed) || (errors.Is(err, os.ErrDeadlineExceeded) && r.ring.isEmpty()) {
```

Comment:
Seems like we now ignore errors that are not `ErrClosed` or `ErrDeadlineExceeded`? I'd prefer to keep returning unknown errors. The if condition is also a bit hard to read. How about splitting it up like so:

```suggestion
			if errors.Is(err, os.ErrDeadlineExceeded) && !r.ring.isEmpty() {
				// TODO: Explain why we're ignoring this
				err = nil
			}
			if err != nil {
```